docs/management/advanced-options.asciidoc

```diff
@@ -83,6 +83,10 @@ specific dashboard, application, or saved object as they enter each space.
 [[fields-popularlimit]]`fields:popularLimit`::
 The top N most popular fields to show.
 
+[[fileupload-maxfilesize]]`fileUpload:maxFileSize`::
+Sets the file size limit when importing files. The default
+value is `100MB`. The highest supported value for this setting is `1GB`.
```

Comment:
the 1GB is not validated in the UX when actually setting this, allowing the user to put in a value that exceeds this. Is it worth to still preserve this 1GB limit? If so, is it possible for the UX to reject an invalid configuration? Due to the client-side batching, it pushes the responsibility to the client, which makes this limit seem artificial. (but maybe this was discussed earlier).

Reply:
I don't think there is a way to validated the advanced setting configuration when its set. I am just using the existing code from ml so that part did not change, other then moving to file_upload plugin. The value is validated against the 1GB when read and will be 1GB if the user provided a larger value. I would consider advanced setting validation out of scope for this PR since that is existing code that did not change functionality, just its name.

Reply:
I agree, if we're streaming it's worth asking why there's a limit at all.

I assume it's to prevent users from uploading huge data sets that end up increasing storage costs or maybe even causing Elasticsearch to run out of disk space. If that's the case, I wonder if this shouldn't rather be a configuration option that only the system administrators can change instead of a ui setting that any user can bump.